File: entity/ExamResponse.py
class ExamResponse:
    def __init__(self, question_number, student_answer, confidence, is_correct, score, sub_question_number=None):
        self.question_number = question_number
        self.student_answer = student_answer
        self.confidence = confidence
        self.is_correct = is_correct
        self.score = score
        self.sub_question_number = sub_question_number  # 새끼 문제 번호 (optional)

    def to_dict(self):
        return {
            "question_number": self.question_number,
            "student_answer": self.student_answer,
            "confidence": self.confidence,
            "is_correct": self.is_correct,
            "score": self.score,
            "sub_question_number": self.sub_question_number  # 새끼 문제 번호 포함
        }

File: checkmate/BACKEND/src/main/java/com/checkmate/ai/entity/ExamResponse.java
package com.checkmate.ai.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;


@Getter
@Setter
@Entity
public class ExamResponse {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long examResponseId;  // 데이터베이스에서 식별할 ID
    private int questionNumber;
    private int subQuestionNumber;
    private String studentAnswer;
    private int answerCount;
    private float confidence;
    private boolean isCorrect;
    private float score;

//
//    // ✅ 외래 키 설정
//    @ManyToOne(fetch = FetchType.LAZY)
//    @JoinColumn(name = "student_response_id")  // 이 컬럼이 foreign key가 됨
//    private StudentResponse studentResponse;



}
